ldap/cache.go

```diff
@@ -112,6 +113,27 @@ func (c *userCache) Get(token string) (gimlet.User, bool, error) {
 	return u.user, true, nil
 }
 
+func (c *userCache) Clear(u gimlet.User, all bool) error {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	if all {
+		c.cache = make(map[string]cacheValue)
+		c.userToToken = make(map[string]string)
+		return nil
+	}
+
+	token, ok := c.userToToken[u.Username()]
+	if !ok {
+		return errors.New("invalid user")
```

Comment:
Won't this always return an error in the `all == true` case?

Reply:
If all == true this code will not be reached (it will return on 123)